testapp1.py

```diff
@@ -1,68 +1,83 @@
 import os
 import uuid
 from datetime import timedelta
-from flask import Flask, redirect, request, session, url_for
+
 import jwt
+from quart import Quart, redirect, request, session, url_for
+
 from generate_jwt import generate_jwt
 
-app = Flask(__name__)
+app = Quart(__name__)
 
 # Generate a secret key for the session cookie
-SECRET_KEY = os.getenv('SECRET_KEY', str(uuid.uuid4()))
+SECRET_KEY = os.getenv("MEMBRANE_SECRET_KEY", str(uuid.uuid4()))
 app.secret_key = SECRET_KEY
 
-session_lifetime_minutes = int(os.environ.get('SESSION_LIFETIME_MINUTES', 30))
-app.config['PERMANENT_SESSION_LIFETIME'] = timedelta(minutes=session_lifetime_minutes)
+session_lifetime_minutes = int(os.getenv("MEMBRANE_SESSION_LIFETIME_SECONDS", 30))
+app.config["PERMANENT_SESSION_LIFETIME"] = timedelta(minutes=session_lifetime_minutes)
+
 
 def decode_jwt_token(jwt_token, public_key):
     """
     Decoding the JWT using the given public key.
     """
     try:
-        decoded_token = jwt.decode(jwt_token, public_key, algorithms=['RS256'])
+        decoded_token = jwt.decode(jwt_token, public_key, algorithms=["RS256"])
         return decoded_token
     except Exception as error:
         print(f"Error decoding JWT: {error}")
         return None
 
-@app.route('/')
-def hello_world():
+
+@app.route("/health")
+async def health_check():
+    return "ok", 200
+
+
+@app.route("/")
+async def hello_world():
     # Check if there's a session cookie
-    if 'authenticated' in session and session['authenticated']:
-        email = session['decoded_token']['sub']  # Get the email from the decoded token
-        return f'Hello, {email}!'
+    if "authenticated" in session and session["authenticated"]:
+        email = session["decoded_token"]["sub"]  # Get the email from the decoded token
+        return f"Hello, {email}!"
 
     # Check if there's a JWT token in the request from another backend
-    jwt_token = request.args.get('token')
+    jwt_token = request.args.get("token")
     if not jwt_token:
         # Generate JWT token
-        with open('tests/client_private_keys/testapp1_private_key.pem', 'rb') as file:
+        with open("keys/testapp1_private_key.pem", "rb") as file:
             private_key_content = file.read()
         data = {
             "app_id": "testapp1",
-            "redirect_url": url_for("hello_world", _external=True)
+            "redirect_url": url_for("hello_world", _external=True),
         }
-        jwt_token = generate_jwt(data, private_key_content)
+        jwt_token = generate_jwt(
+            data,
+            private_key_content,
+            {"alg": "RS256", "typ": "JWT", "app_id": "testapp1"},
+        )
         # Redirect to /authenticate with the generated JWT token
         print("Generate JWT token AND SENT OUT")
         return redirect(f"http://127.0.0.1:5000/authenticate?token={jwt_token}")
```

Comment:
should be able to build hostname and port from app container or request information instead of building something that only works locally on port 5000.

Reply:
need clarification. Also this is only the test app. Isn't it supposed to be removed from the project later?

Reply:
as discussed, this is actually in the client test app that we should spin off to flask-example and have a MEMBRANE_BACKEND_URL environment variable.